### PR TITLE
Add page with information about GSoC 2018

### PR DESCRIPTION
**Pull request by rdowner**
Add a page to the "Community" section which describes Google Summer of Code 2018, and answers some frequently asked questions.

### CODE REVIEW

**Comment by rdowner** [comment]
@tbouron would appreciate it if you could give this a review, since you have also been interacting with potential students - is there anything else that should be added here? Thanks!

File: community/gsoc.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Thank you for considering Apache Brooklyn and The Apache Software Foundation for
+your GSoC project. Welcome to our community!
+
+Firstly, if you have not already done so, please read the [Student Guide to
+Google Summer of Code](https://google.github.io/gsocguides/student/). It
+contains essential information about how to approach your project before and
+after your application is submitted. Also, please read the [GSoC information
+provided by The Apache Software Foundation](https://community.apache.org/gsoc.html),
+too.
```
Comment (tbouron) [inline community/gsoc.md]: Can drop the `too`don't you think? (as well as the coma that come before it)

File: community/gsoc.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Secondly, Apache values community, and the community and all its work is done in
+the open, on public mailing lists and IRC channels. Please [subscribe to our
+mailing lists](https://brooklyn.apache.org/community/mailing-lists.html) and
```
Comment (tbouron) [inline community/gsoc.md]: This you can use relative links for this?

File: community/gsoc.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+optionally [join our IRC channel](http://localhost:4000/community/irc.html).
```
Comment (tbouron) [inline community/gsoc.md]: Should be either relative link or `brooklyn.apache.org`

File: community/gsoc.md (hunk 1 above)
Comment (rdowner) [reply community/gsoc.md]: 👍

File: community/gsoc.md (hunk 2 above)
Comment (rdowner) [reply community/gsoc.md]: 👍

File: community/gsoc.md (hunk 3 above)
Comment (rdowner) [reply community/gsoc.md]: 👍

**Comment by rdowner** [comment]
Thanks @tbouron - I'll make the suggested changes and then merge.

**Pull request merged**